x-pack/legacy/plugins/uptime/public/pages/page_header.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiTitle, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import React, { useEffect, useState, useContext } from 'react';
+import { connect } from 'react-redux';
+import { useRouteMatch } from 'react-router-dom';
+import { i18n } from '@kbn/i18n';
+import { UptimeDatePicker, CommonlyUsedRange } from '../components/functional/uptime_date_picker';
+import { AppState } from '../state';
+import { selectMonitorStatus } from '../state/selectors';
+import { getMonitorPageBreadcrumb, getOverviewPageBreadcrumbs } from '../breadcrumbs';
+import { stringifyUrlParams } from '../lib/helper/stringify_url_params';
+import { useUrlParams } from '../hooks';
+import { UptimeSettingsContext } from '../contexts';
+import { getTitle } from '../lib/helper/get_title';
+import { UMUpdateBreadcrumbs } from '../lib/lib';
+
+interface PageHeaderProps {
+  commonlyUsedRanges?: CommonlyUsedRange[];
+  monitorStatus?: any;
+  setBreadcrumbs: UMUpdateBreadcrumbs;
+}
+
+export const PageHeaderComponent = ({
+  commonlyUsedRanges,
+  monitorStatus,
+  setBreadcrumbs,
+}: PageHeaderProps) => {
+  const monitorPage = useRouteMatch({
+    path: '/monitor/:monitorId/:location?',
+  });
+  const { refreshApp } = useContext(UptimeSettingsContext);
+
+  const [getUrlParams] = useUrlParams();
+  const { absoluteDateRangeStart, absoluteDateRangeEnd, ...params } = getUrlParams();
+
+  const headingText = i18n.translate('xpack.uptime.overviewPage.headerText', {
```

Comment:
It's a bit odd to me that this component has values for its title contained within it. Have you considered having the page that embeds this inject either the title or a function that generates the title?

Reply:
There is room for improvement here, i will insert a function.